deeppavlov/models/preprocessors/ubuntu_preprocessor.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2018 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from typing import List, Union, Iterable
+
+from deeppavlov.core.commands.utils import expand_path
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.models.component import Component
+from deeppavlov.core.models.estimator import Estimator
+from deeppavlov.core.data.utils import zero_pad_truncate
+
+log = get_logger(__name__)
+
+
+@register('ubuntu_preprocessor')
+class UbuntuPreprocessor(Estimator):
```

Comment:
There are very few differences between this class and `SiameseReader` class. Could you please instead add this functionality to the `SiameseReader` using some flags?

Reply:
`SiamesePreprocessor`, you mean?
OK, I will merge the code.

Reply:
Yes, exactly. Thank you!

Reply:
Merged.
When `sent_vocab` is not used (i.e. when there is not `sent_vocab` argument in a config), it should not be fit. Is it good to write like:
https://github.com/deepmipt/DeepPavlov/blob/8c9e9fd176abbb799b075512d10b1d7effa2ed17/deeppavlov/models/preprocessors/siamese_preprocessor.py#L103?